src/Reflector/Generic.php

```diff
@@ -34,10 +35,20 @@ public function __construct(SourceLocator $sourceLocator)
      */
     public function reflect(Identifier $identifier)
     {
-        return $this->reflectFromLocatedSource(
-            $identifier,
-            $this->sourceLocator->__invoke($identifier)
-        );
+        $aggregate = $this->sourceLocator;
+        if (!($aggregate instanceof AggregateSourceLocator)) {
+            $aggregate = new AggregateSourceLocator([$this->sourceLocator]);
+        }
+
+        foreach ($aggregate->__invoke($identifier) as $locatedSource) {
```

Comment:
`$aggregate($identifier)`. If you use `__invoke` explicitly, you lose the advantage of type-hinting `callable`